packages/nimble-components/src/table/testing/table.pageobject.ts

```diff
@@ -570,52 +563,31 @@ export class TablePageObject<T extends TableRecord> {
         columnIndex: number,
         deltas: readonly number[]
     ): void {
-        const divider = this.getColumnLeftDivider(columnIndex);
-        if (!divider) {
-            throw new Error(
-                'The provided column index has no left divider associated with it.'
-            );
-        }
-        const dividerRect = divider.getBoundingClientRect();
-        let currentMouseX = (dividerRect.x + dividerRect.width) / 2;
-        const mouseDownEvent = new MouseEvent('mousedown', {
-            clientX: currentMouseX,
-            clientY: (dividerRect.y + dividerRect.height) / 2
-        });
-        divider.dispatchEvent(mouseDownEvent);
-
-        for (const delta of deltas) {
-            currentMouseX += delta;
-            const mouseMoveEvent = new MouseEvent('mousemove', {
-                clientX: currentMouseX
-            });
-            document.dispatchEvent(mouseMoveEvent);
-        }
-
-        const mouseUpEvent = new MouseEvent('mouseup');
-        document.dispatchEvent(mouseUpEvent);
+        this.dragSizeColumnByDivider(columnIndex, true, deltas);
     }
 
-    public getColumnRightDivider(index: number): HTMLElement | null {
-        const headerContainers = this.tableElement.shadowRoot!.querySelectorAll('.header-container');
-        if (index < 0 || index >= headerContainers.length) {
-            throw new Error(
-                'Invalid column index. Index must be greater than or equal to 0 and less than the number of visible columns.'
-            );
-        }
-
-        return headerContainers[index]!.querySelector('.column-divider.right');
+    /**
+     * @param columnIndex The index of the column whose right divider we're checking.
+     */
+    public columnRightDividerHasActiveStyling(columnIndex: number): boolean {
+        const divider = this.getColumnDivider(columnIndex, false);
+        const currentColor = parseColor(
+            window.getComputedStyle(divider).borderColor
+        );
+        const hoverColor = parseColor(borderHoverColor.getValueFor(divider))!;
+        return currentColor?.equalValue(hoverColor) ?? false;
```

Comment:
Maybe this should be a non-null assertion. In tests I think it's better to fail faster and I'm guessing there is no circumstance we expect this to every be nulliish.


```suggestion
        return currentColor!.equalValue(hoverColor);
```